### PR TITLE
Add auto rebase bot

### PR DESCRIPTION
**Pull request by fortianyou**

### CODE REVIEW

File: scripts/ci/auto_rebase.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+date_str=$(date '+%Y%m%d')
+rebase_branch=features/bot_aicompiler_rebase_${date_str}
+commit_msg="[BOT] aicompiler rebase ${date_str}"
```
Comment (Yancey1989) [inline scripts/ci/auto_rebase.sh:15]: bladedisc rebase?

File: scripts/ci/auto_rebase.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  echo ${BOT_GITHUB_TOKEN} | gh auth login --with-token
+  gh pr create -B main -R ${remote_repo} -H ${rebase_branch} \
+     --reviewer "fortianyou,qiuxiafei,wyzero,Yancey1989" \
+     --title "${commit_msg}" \
+     --body "${tf_commit_body}"
+}
+
+cd tf_community && rebase_tf && cd ..
+create_pr
```
Comment (Yancey1989) [inline scripts/ci/auto_rebase.sh:47]: This bot only detect the rebase conflict, maybe we need to run all the test jobs to evaluate the rebased code.

Comment (fortianyou) [reply scripts/ci/auto_rebase.sh:47]: The CI check will not run in this BOT. This BOT will create a new PR. If the PR CI checked passed, we could merge it directly.

Comment (Yancey1989) [reply scripts/ci/auto_rebase.sh:47]: I see, thanks.

**Review by Yancey1989 — APPROVED** [review]
LGTM

File: scripts/ci/auto_rebase.sh (hunk 1 above)
Comment (fortianyou) [reply scripts/ci/auto_rebase.sh:15]: I will update the commit msg later

**Pull request merged**